qiskit/providers/aer/backends/cluster/clusterjobset.py

```
@@ -0,0 +1,236 @@
+                raise AerClusterTimeoutError(
+                    'Timeout while waiting for the results of experiment {}'.format(
+                        cjob.name()))
+            if timeout:
+                timeout = original_timeout - (time.time() - start_time)
+                if timeout <= 0:
+                    raise AerClusterTimeoutError(
+                        "Timeout while waiting for JobSet results")
+
+        self._results = CResults(self, res)
+
+        return self._results
+
+    @requires_submit
+    def cancel(self) -> None:
+        """Cancel all jobs in this job set."""
+        for cjob in self._futures:
+            cjob.cancel()
+
+    @requires_submit
+    def job(self, experiment: Union[str, QuantumCircuit, Schedule]) -> Tuple[CJob, int]:
+        """Retrieve the job used to submit the specified experiment and its index.
+
+        Args:
+            experiment: Retrieve the job used to submit this experiment. Several
+                types are accepted for convenience:
+
+                    * str: The name of the experiment.
+                    * QuantumCircuit: The name of the circuit instance will be used.
+                    * Schedule: The name of the schedule instance will be used.
+
+        Returns:
+            A tuple of the job used to submit the experiment and the experiment index.
+
+        Raises:
+            AerClusterJobNotFound: If the job for the experiment could not
+                be found.
+        """
+        if isinstance(experiment, (QuantumCircuit, Schedule)):
+            experiment = experiment.name
+        for job in self.jobs():
+            for i, exp in enumerate(job.qobj().experiments):
+                if hasattr(exp.header, 'name') and exp.header.name == experiment:
+                    return job, i
+
+        raise AerClusterJobNotFound(
+            'Unable to find the job for experiment {}.'.format(experiment))
+
+    @requires_submit
+    def jobs(self) -> List[Union[CJob, None]]:
```

See above comment about `worker_job` function. If a worked job failed to submit I think it should raise an exception rather than return None.